### PR TITLE
fix(webhook): persist injected connection type to survive secret deletion

### PR DESCRIPTION
**Pull request by mlassak**
## Description
 - Fix connection webhook stripping `storageUri` when the old connection secret is deleted before the InferenceService update
  - Persist the injected connection type as an annotation (`opendatahub.io/injected-connection-type`) on the InferenceService during injection, so cleanup can determine the correct type-specific path even when the
  original secret is gone

**The same fix was needed on the opendatahub-operator side**: https://github.com/opendatahub-io/opendatahub-operator/pull/3748

JIRA ref: [RHOAIENG-61105](https://redhat.atlassian.net/browse/RHOAIENG-61105)

## How Has This Been Tested?
Manually tested the scenario described in the ticket on an OpenShift cluster

## Merge criteria:

- [x] JIRA(s) are linked in the PR description
- [x] The commits are squashed in a cohesive manner and have meaningful messages.
- [x] Testing instructions have been added in the PR body (for PRs involving changes that are not immediately obvious).
- [x] The developer has manually tested the changes and verified that the changes work

## Summary by CodeRabbit

* **New Features**
  * Added an annotation-based lifecycle marker to persist the injected connection type on inference services for more reliable later cleanup.
  * Webhook injection/removal now updates this marker for create, replacement, and removal flows.

* **Bug Fixes**
  * Update handling now restores the previous connection type from the marker when the old connection secret is missing.
  * Validation now fails when the referenced secret lacks a supported connection protocol annotation.

* **Tests**
  * Expanded coverage for annotation lifecycle and scoped re-injection across S3, OCI, and URI scenarios (including replacement edge cases).

[RHOAIENG-61105]: https://redhat.atlassian.net/browse/RHOAIENG-61105?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Comment by carlkyrillos** [comment]
@israel-hdez Can you review this now that the webhook is in the component repo? Thanks!

File: internal/webhook/serving/v1alpha2/llminferenceservice_webhook_test.go [hunk 1]
```diff
@@ -890,6 +890,128 @@ var _ = Describe("LLMInferenceService ConnectionsAPI Defaulter", func() {
...
+			}
+			preURI, parseErr := apis.ParseURL("oci://registry.example.com/model:latest")
+			Expect(parseErr).ToNot(HaveOccurred())
+			newLLM.Spec.Model.URI = *preURI
+			admCtx := llmAdmissionCtx(admissionv1.Update, oldLLM, false)
+
+			Expect(d.Default(admCtx, newLLM)).To(Succeed())
+			// Phase 2 of cleanup unconditionally zeroes any previously-set model URI,
+			// and OCI injection does not currently populate it, so it stays empty.
+			Expect(newLLM.Spec.Model.URI.String()).To(BeEmpty())
```
Comment (israel-hdez) [inline internal/webhook/serving/v1alpha2/llminferenceservice_webhook_test.go]: This comment is true... but isn't it what the we should be preventing to happen?

In contrast, the ISVC case looks right.

Comment (mlassak) [reply internal/webhook/serving/v1alpha2/llminferenceservice_webhook_test.go]: Should be fixed now

**Review by israel-hdez — APPROVED** [review]
/lgtm
